File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res: int = 0
        l_idx = 0
        r_idx = 1
        while r_idx < len(prices):
            if prices[l_idx] < prices[r_idx]:
                profit = prices[r_idx] - prices[l_idx]
                res = max(res, profit)
            else:
                l_idx = r_idx
            r_idx += 1
        return res

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
      int running_min = 1e9;
      int best_profit = 0;

      for (int price: prices){
        int best_selling_here = price - running_min;
        best_profit = max(best_profit, best_selling_here);
        running_min = min(running_min, price);
      }
      return best_profit;
    }
};
